### PR TITLE
SLCLI-15 Support connected mode

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/cli/analysis/DefaultLogOutput.java [hunk 1]
```diff
@@ -17,25 +17,29 @@
...
  */
-package org.sonarlint.cli;
+package org.sonarlint.cli.analysis;
 
 import org.sonarlint.cli.util.Logger;
 import org.sonarsource.sonarlint.core.client.api.common.LogOutput;
 
 class DefaultLogOutput implements LogOutput {
 
   private final Logger logger;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/analysis/DefaultLogOutput.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: src/main/java/org/sonarlint/cli/config/ConfigurationReader.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  private String getContents(Path filePath) {
+    try {
+      return Files.toString(filePath.toFile(), Charsets.UTF_8);
+    } catch (IOException e) {
+      throw new IllegalStateException("Error reading configuration file: " + filePath.toAbsolutePath(), e);
+    }
+  }
+
+  private ProjectConfiguration validate(ProjectConfiguration projectConfig, Path configFilePath) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/config/ConfigurationReader.java]: MAJOR Make "validate" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: src/main/java/org/sonarlint/cli/config/ConfigurationReader.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if (StringUtils.isEmpty(projectConfig.projectKey())) {
+      throw new IllegalStateException("Project binding must have a project key defined. Check the configuration in: " + configFilePath.toAbsolutePath());
+    }
+    return projectConfig;
+  }
+
+  private GlobalConfiguration validate(GlobalConfiguration globalConfig, Path configFilePath) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/config/ConfigurationReader.java]: MAJOR Make "validate" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: src/main/java/org/sonarlint/cli/config/ConfigurationReader.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    String contents = getContents(configFilePath);
+    return validate(new GsonBuilder().create().fromJson(contents, GlobalConfiguration.class), configFilePath);
+  }
+
+  public ProjectConfiguration readProject(Path configFilePath) {
+    String contents = getContents(configFilePath);
+    return validate(new GsonBuilder().create().fromJson(contents, ProjectConfiguration.class), configFilePath);
+  }
+
+  private String getContents(Path filePath) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/config/ConfigurationReader.java]: MAJOR Make "getContents" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: src/main/java/org/sonarlint/cli/analysis/ConnectedSonarLint.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      .orElseThrow(() -> new IllegalStateException("Module key '" + moduleKey + "' not found in the SonarQube server"));
+    updateModule();
+    LOGGER.info("Binding updated");
+  }
+
+  private void updateModule() {
+    engine.updateModule(getServerConfiguration(server), moduleKey);
+  }
+
+  private ServerConfiguration getServerConfiguration(SonarQubeServer server) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/analysis/ConnectedSonarLint.java]: MAJOR Make "getServerConfiguration" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarlint/cli/config/ConfigurationReader.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.cli.config;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.sonar.api.internal.apachecommons.lang.StringUtils;
```
Comment (henryju) [inline src/main/java/org/sonarlint/cli/config/ConfigurationReader.java]: Wrong import ?

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
Import issue then LGTM

**Pull request merged**